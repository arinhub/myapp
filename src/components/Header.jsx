import React, {memo, useState, useEffect} from 'react';
import { useSelector, shallowEqual} from 'react-redux';
import {searchedResult} from '../actions';
import Search from './Search';

const Header = () => {
    const searchedData = useSelector(searchedResult, shallowEqual);
    const [isSearchDataAvailable, setIsSearchDataAvailable] = useState(false);
    const [back, goBack] = useState(false);

    useEffect(() => {
        if (searchedData.length > 0) {
            setIsSearchDataAvailable(true)
        } else {
            setIsSearchDataAvailable(false)
        }
    }, [searchedData]);

    const handleClick = () => {
        if (!isSearchDataAvailable) return false;
        goBack(true);
    }

    return (
        <div className={'nav-bar'}>
            <div className={'back-button-container'}>
                <img src={process.env.PUBLIC_URL + '/assets/Back.png'} 
                alt='back' 
                className={`back-button ${!isSearchDataAvailable && 'blur'}`}
                onClick={handleClick}
                />
            </div>
            <Search handleBack={back}/>
        </div>
    );
};

export default memo(Header)