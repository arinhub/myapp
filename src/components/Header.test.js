import renderer from 'react-test-renderer';
import React from 'react';
import Header from './Header';

const mockDispatchFunction = jest.fn();

jest.mock('react-redux', () => {
  return {
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn().mockImplementation(selector => selector()),
    useDispatch: () => {
      return mockDispatchFunction;
    },
  };
});

it('renders correctly', () => {
  const tree = renderer.create(
    <Header />
  );
  expect(tree).toMatchSnapshot();
});
