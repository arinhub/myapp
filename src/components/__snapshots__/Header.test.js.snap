// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="nav-bar"
>
  <div
    className="back-button-container"
  >
    <img
      alt="back"
      className="back-button blur"
      onClick={[Function]}
      src="/assets/Back.png"
    />
  </div>
  <div
    className="search_field"
  >
    <input
      name="query"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Search movies!"
      value=""
    />
    <button
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Search"
        className="search-icon"
        src="/assets/search.png"
      />
    </button>
    <div
      className="Toastify"
    />
  </div>
</div>
`;
